Source/ImagePickerController.swift

```diff
@@ -1,16 +1,243 @@
+    galleryView.frame.origin.y = initialFrame.origin.y + translation.y
+    galleryView.topSeparator.frame.origin.y = 0
+
+    if galleryView.frame.size.height - galleryView.topSeparator.frame.height > 100 {
+      galleryView.collectionViewLayout.invalidateLayout()
+      galleryView.collectionView.frame.size.height = galleryView.frame.size.height - galleryView.topSeparator.frame.height
+      galleryView.collectionSize = CGSizeMake(galleryView.frame.size.height - galleryView.topSeparator.frame.height, galleryView.frame.size.height - galleryView.topSeparator.frame.height)
+      galleryView.collectionView.reloadData()
+    } else {
+      galleryView.collectionView.frame.origin.y = galleryView.topSeparator.frame.height
+    }
+
+    if location.y >= initialFrame.origin.y + initialFrame.height - galleryView.topSeparator.frame.height {
+      galleryView.frame.size.height = galleryView.topSeparator.frame.height
+      galleryView.frame.origin.y = initialFrame.origin.y + initialFrame.height - galleryView.topSeparator.frame.height
+    } else if galleryView.collectionView.frame.height >= ImageGalleryView.Dimensions.galleryHeight {
+      galleryView.frame.size.height = ImageGalleryView.Dimensions.galleryHeight + galleryView.topSeparator.frame.height
+      galleryView.frame.origin.y = initialFrame.origin.y + initialFrame.height - galleryView.topSeparator.frame.height - ImageGalleryView.Dimensions.galleryHeight
+      galleryView.collectionView.frame.size.height = ImageGalleryView.Dimensions.galleryHeight
+      galleryView.collectionSize = CGSizeMake(galleryView.collectionView.frame.height, galleryView.collectionView.frame.height)
+      galleryView.collectionView.reloadData()
+    }
+
+    cameraController.view.frame.size.height = galleryView.frame.origin.y - topView.frame.height
+    cameraController.view.frame.origin.y = topView.frame.height
+  }
+
+  func panGestureDidEnd(translation: CGPoint, location: CGPoint, velocity: CGPoint) {
+    if galleryView.frame.height < 134 && velocity.y < 0 {
+      UIView.animateWithDuration(0.2, animations: { [unowned self] in
+        self.galleryView.frame.size.height = 134
+        self.galleryView.frame.origin.y = self.initialFrame.origin.y + self.initialFrame.height - self.galleryView.topSeparator.frame.height - 100
+        self.galleryView.collectionViewLayout.invalidateLayout()
+        self.galleryView.collectionView.frame.size.height = 134 - self.galleryView.topSeparator.frame.height
+        self.galleryView.collectionSize = CGSizeMake(self.galleryView.collectionView.frame.height, self.galleryView.collectionView.frame.height)
+        self.cameraController.view.frame.size.height = self.galleryView.frame.origin.y - self.topView.frame.height
+        self.cameraController.view.frame.origin.y = self.topView.frame.height
+        }, completion: { finished in
+          self.galleryView.collectionView.reloadData()
+      })
+    } else if velocity.y < -100 {
```

Comment:
These velocity checks may cause blinks when you drag handle very fast. This is actually a common usage pattern, as if you try to drag with a finger on a  real device, the velocity of 100 is not that big.

Please, check it. I've put a bug demo here: http://youtu.be/8ixEVzvH43A